src/phy/tuntap_interface.rs

```diff
@@ -37,6 +37,19 @@ impl TunTapInterface {
             medium,
         })
     }
+
+    /// Attaches to a TUN/TAP interface specified by file descriptor `fd`.
+    ///
+    /// On platforms like Android, a file descriptor to a tun interface is exposed.
+    /// On these platforms, a TunTapInterface cannot be instantiated with a name.
+    pub fn from_fd(fd: RawFd, medium: Medium, mtu: usize) -> io::Result<TunTapInterface> {
```

Comment:
why set the MTU manually? `SIOCGIFMTU` doesn't work on Android?

Reply:
Because the device name is not known when using the [VpnService builder](https://developer.android.com/reference/android/net/VpnService.Builder), so I wouldn't know the device name to supply as part of the `ifreq` struct to obtain the MTU through `SIOCGIFMTU`.

Reply:
aw, that's unfortunate. Oh well.